### ISSUE TITLE
Stepper - Dev

### ISSUE THREAD

**Issue by iamskip** [GemeenteNijmegen/component-library#249]

**Comment by mwbankers87** [comment]
@MariannevanLierop @vladimirdacosta Dit component is exact hetzelfde als de Step-list die al bestaat.
Ook de tekst in de beschrijving is hetzelfde. 

Ik wilde dus informeren wat exact de bedoeling is. Moet er een exacte kopie komen die 'Proces steps' heet?
Dan krijgen we duplicated code. De vraag is of dat wenselijk is.

**Comment by vladimirdacosta** [comment]
@mwbankers87 ik had de verkeerd link. Het gaat hier om de progress listen en dat is een andere component dan de Step-list

**Comment by mwbankers87** [comment]
@vladimirdacosta Duidelijk! Bedankt voor de snelle reactie.

**Comment by mwbankers87** [comment]
@vladimirdacosta Ik heb een begin gemaakt aan dit component, maar de tokens lijken er nog niet in te zitten.
Volgens mij staat er ook nog een PR voor.

**Comment by vladimirdacosta** [comment]
@mwbankers87 dit ga ik deze week nog met @Rozerinay bespreken.

**Comment by MariannevanLierop** [comment]
Omdat Rozerin op vakantie is kunnen we de tokens nog niet voor maandag aanleveren, dit ticket blijft dus nog even staan.

**Comment by MariannevanLierop** [comment]
@vladimirdacosta , graag update over tokens zodra je dit weet

**Comment by MariannevanLierop** [comment]
@mwbankers87, zie https://app.slack.com/client/T68FXPFQV/C088PMZC90X/1780495887.293779

**Comment by mwbankers87** [comment]
@MariannevanLierop @vladimirdacosta Het Progress list component is toegevoegd aan de [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/docs/web-component-progresslist--docs)

**Comment by vladimirdacosta** [comment]
@mwbankers87 Ziet er goed uit! Ik mis alleen nog de implementatie van _step-header en _step-meta (als paragraph). Die zie ik op dit moment nog niet terug. Kun je die nog toevoegen? Dan is de implementatie compleet.

<img width="892" height="339" alt="Image" src="https://github.com/user-attachments/assets/5cff3afb-2fd2-430b-abbc-eadfbdf2d98d" />

**Comment by mwbankers87** [comment]
@vladimirdacosta Er is een mogelijkheid om tekst als paragraph toe te voegen (stap 4 in storybook heeft er een), maar ik had even gemist dat er 2 stijlen voor waren.

Maar deze kunnen nog toegevoegd worden, ik zal er naar kijken.

**Comment by vladimirdacosta** [comment]
@mwbankers87 Super, dank je wel. Dan wacht ik de aanvulling af.

**Comment by mwbankers87** [comment]
@vladimirdacosta Ik heb de meta-date toegevoegd. Ook heb ik nog 2 voorbeeldjes toegevoegd waarbij alles geopend en alles gesloten is. Per stap is de gewenste functionaliteit te bepalen.

**Comment by vladimirdacosta** [comment]
@mwbankers87 Dank voor de aanvulling! Ik mis nog één variant. Bij stap 4 zie ik nu de optie om alleen een paragraaf onder de stap te tonen (zonder bullets). Diezelfde mogelijkheid zou ik ook graag zien bij een afgeronde stap, zoals stap 1 of 2. Nu lijken afgeronde stappen alleen de variant met bullets te ondersteunen.

**Comment by mwbankers87** [comment]
@vladimirdacosta Dat is gewoon mogelijk. Ik heb nu de voorbeelden uit Figma aangehouden, daar stond geen tekst bij de substappen. Maar ik zal bij stap 1 en 2 een tekst toevoegen.

**Comment by vladimirdacosta** [comment]
@mwbankers87 Volgens mij hebben we het over verschillende varianten. In Figma staat op de Step Body-pagina onder True al een afgeronde stap met body uitgewerkt. Die body hoeft niet alleen uit bullets te bestaan; deze moet ook een losse paragraaf kunnen bevatten. Die variant mis ik nog in Storybook.

**Comment by mwbankers87** [comment]
@vladimirdacosta Ok, dus een afgeronde stap met enkel een body? 
Ik zit even te kijken hoe ik deze variant kan noemen. Ik kan een versie maken waarbij alles is afgerond, dus stap 4 ook. Is dat een optie?

**Comment by vladimirdacosta** [comment]
@mwbankers87 Bijna. 😊

Het gaat mij niet om een aparte variant “afgeronde stap met alleen een body”. De body is in het ontwerp een optioneel onderdeel van iedere step, ongeacht of die afgerond, actief of nog niet afgerond is.

In de documentatie staat ook:

> “Het is mogelijk om per Step de Body toe te voegen. Je kunt ook ervoor kiezen om afzonderlijk de Meta en Meta date in de Body te laten zien of te verbergen.”

Voor mij hoeft dit dus geen aparte variant te zijn, maar een eigenschap van de Step. Storybook zou idealiter moeten laten zien dat de Body bij elke status van een Step toegevoegd kan worden (met of zonder bullets, Meta en Meta date).

**Comment by mwbankers87** [comment]
@vladimirdacosta Met variant bedoel ik puur hoe ik het binnen storybook laat zien in de sidebar, niet een variant op de stap:

<img width="245" height="310" alt="Image" src="https://github.com/user-attachments/assets/55b7601d-1864-4242-9d7f-639696bc216d" />

Het is zo gebouwd exact als je het beschrijft. Elke stap heeft een Status (dus stijl), Header, Body, en substeps. Dit zijn natuurlijk niet allemaal verschillende objecten, maar eigenschappen van dezelfde stap. Als je iets niet invuld, word het niet getoond.

Er is wel een mogelijkheid om de eigenschappen van een stap te zien en te veranderen binnen Storybook. Hier zie je wel goed dat het om het zelfde component gaat, met een andere eigenschap.:
https://github.com/user-attachments/assets/b1ba1c84-e168-481d-bd76-61ece764a488

**Comment by mwbankers87** [comment]
@vladimirdacosta Ik heb een voorbeeld toegevoegd waar alles, inclusief stap 4 checked is: [Zie storybook](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-progresslist--checked)

Ik heb in dit geval dus enkel de status van stap 3 en stap 4 op "checked" gezet.

**Comment by vladimirdacosta** [comment]
@mwbankers87  Volgens mij bedoelen we iets anders. 😊

Het gaat mij niet om de status van de step of de properties van de component. Mijn punt gaat over de inhoud van de Step Body.

In de uitgeklapte stappen zie ik nu alleen voorbeelden waarbij de body uit een lijst (bullets) bestaat. Volgens het ontwerp kan de body echter ook uit een gewone paragraaf bestaan, zonder bullets.

Ik zou daarom graag ook een Storybook-voorbeeld zien waarin een uitgeklapte stap alleen een paragraaf bevat. Dan is duidelijk dat de Step Body beide vormen ondersteunt: een paragraaf óf een lijst.

**Comment by mwbankers87** [comment]
@vladimirdacosta Ik heb een nieuw voorbeeld gemaakt, waarbij stap 1 en 2 inplaats van substeps een paragraaf bevatten:
ZIe [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-progresslist--paragraph)

cc @Michel-Verhoeven

**Comment by vladimirdacosta** [comment]
@mwbankers87 Ja, dit is precies wat ik bedoelde. Dank je wel voor het toevoegen! Hij is goed zo!

**Comment by MariannevanLierop** [comment]
@ElsGroffen , zou jij ook de Progress list willen checken op [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-progresslist--paragraph)?

**Comment by ElsGroffen** [comment]
@MariannevanLierop ik zie in de voorbeelden geen gekke dingen qua toegankelijkheid. Wel geeft de github omgeving zelf bij ieder voorbeeld een violation aan , @mwbankers87 graag even een check of deze opmerking terecht is.

**Comment by mwbankers87** [comment]
@ElsGroffen Ik had het ook gezien, in principe klopte het wel, maar Storybook zag niet dat het ging om een list.
Ik heb een aanpassing gemaakt zodat de warning weg is.
